### DISCUSSION TITLE
Works on...

### DISCUSSION THREAD

**Discussion by 007revad** [007revad/Synology_enable_Deduplication#31] [General]
Please comment if this worked for you, or didn't work for you, with the following information:

- Working: yes/no
- NAS model: 
- DSM version:
- SSD brand and model:

**Comment by AdrianDForsyth** [comment]
Working: no
NAS model: RS1219+
DSM version: 7.2.1-69057-3
SSD brand and model: Mostly Ironwolfs (HDD)

Cannot pass go, model or DSM version not supported

**Reply by 007revad** [reply]
Unfortunately models with Avoton CPUs don't have the required files in DSM.

**Comment by TheJACKedViking** [comment]
Working: no
NAS model: DS2422+
DSM version: 7.2.1 Update 4
SSD brand and model: WD Red Pros ( HDD )

Script says its enabled. Data De-duplication time scheduling shows up fine. Option to configure specific volumes does not appear. DSM restart and clearing browser cache did not resolve. All volumes have Analytics enabled.

**Reply by 007revad** [reply]
It looks like maybe something has changed in DSM 7.2.1 Update 4

My DS1821+ that had deduplication enabled before updating to DSM 7.2.1 Update 4 still shows the configure data deduplication option for all volumes.

But my DS720+ did **not** have deduplication enabled before updating to DSM 7.2.1 Update 4 and it's **not** showing the configure data deduplication option for any volumes.

So at least it's not just you and I can reproduce the issue... and hopefully fix it.

**Reply by 007revad** [reply]
I can't see anything different in DSM 7.2.1 Update 4.

Can you tell me what these commands return:
```
grep -c 'SYNO.SDS.StorageUtils.supportBtrfsDedupe&&e.dedup_info.show_config_btn' /var/packages/StorageManager/target/ui/storage_panel.js

grep -c 'SYNO.SDS.StorageUtils.supportBtrfsDedupe,' /var/packages/StorageManager/target/ui/storage_panel.js

synogetkeyvalue /etc.defaults/synoinfo.conf support_btrfs_dedupe

synogetkeyvalue /etc/synoinfo.conf support_btrfs_dedupe

synogetkeyvalue /etc.defaults/synoinfo.conf support_tiny_btrfs_dedupe

synogetkeyvalue /etc/synoinfo.conf support_tiny_btrfs_dedupe
```

**Reply by 3QInc** [reply]
xyz@abc:/$ grep -c 'SYNO.SDS.StorageUtils.supportBtrfsDedupe&&e.dedup_info.show_config_btn' /var/packages/StorageManager/target/ui/storage_panel.js
1
xyz@abc:/$ grep -c 'SYNO.SDS.StorageUtils.supportBtrfsDedupe,' /var/packages/StorageManager/target/ui/storage_panel.js
1
xyz@abc:/$ synogetkeyvalue /etc.defaults/synoinfo.conf support_btrfs_dedupeno
xyz@abc:/$ synogetkeyvalue /etc/synoinfo.conf support_btrfs_dedupe
no
xyz@abc:/$ synogetkeyvalue /etc.defaults/synoinfo.conf support_tiny_btrfs_dedupe
yes
xyz@abc:/$ synogetkeyvalue /etc/synoinfo.conf support_tiny_btrfs_dedupe
yes
xyz@abc:/$

**Reply by 3QInc** [reply]
Thanks for your work and innovation on this.  I'm using this on a 923+ with 8GB ram and update 4 of the DSM.

**Reply by 007revad** [reply]
I was reminded the other day that after running the script it can take up to 20 minutes for the Depuplication menu to become available.

**Comment by Vuego99** [comment]
Working: yes
NAS model: 3622xs+
DSM version: DSM 7.2.1-69057-4
SSD brand and model: Kingston 600M, Hdd synology

Thank you for working on the script

**Comment by t0mm1e** [comment]
Working: yes
    NAS model: DS1823xs+, 64 GB mem
    DSM version: DSM 7.2.1-69057 Update 4
    SSD brand and model: WD201KFGX-68BKJN (WD Red Pro)

It worked before, but I did a full reinstall today, and now I can't get it to work. The scripts says it installs fine, but the menu options to deduplicate does not appear. It might be update 4.
Running the checks mentioned above, this is the result:

grep -c 'SYNO.SDS.StorageUtils.supportBtrfsDedupe&&e.dedup_info.show_config_btn' /var/packages/StorageManager/target/ui/storage_panel.js
0
grep -c 'SYNO.SDS.StorageUtils.supportBtrfsDedupe,' /var/packages/StorageManager/target/ui/storage_panel.js
1
synogetkeyvalue /etc.defaults/synoinfo.conf support_btrfs_dedupe
yes
synogetkeyvalue /etc/synoinfo.conf support_btrfs_dedupe
yes
synogetkeyvalue /etc.defaults/synoinfo.conf support_tiny_btrfs_dedupe
no
synogetkeyvalue /etc/synoinfo.conf support_tiny_btrfs_dedupe
no

**Reply by t0mm1e** [reply]
Woah, the menu option just appeared!

Took a while, but it works now 😉

**Reply by 007revad** [reply]
I've noticed it can take up to 20 minutes before the menu appears.

**Comment by cuter-doc0** [comment]
Working: no
NAS model: DS1621+
DSM version: DSM 7.2.1-69057 Update 3
SSD brand and model: TEAM TM8FP6512G
Setup: 6x HDD SHR with 2x nvme SSD cache Btrfs volume

![Screenshot 2024-04-23 at 1 03 18 PM](https://github.com/007revad/Synology_enable_Deduplication/assets/124633877/92f51165-01bf-4a29-9a76-f840dfc9da32)

**Reply by cuter-doc0** [reply]
reading the replies here I might just need to wait - will check back in a bit and report on the result :)

**Comment by 007revad** [comment]
DSM does not allow deduplication of SSD caches. Deduplication is only available for Btrfs volumes.

If volume 1 is using btrfs, and has usage detail analysis enabled, check if the deduplication option is available for volume 1.

**Reply by cuter-doc0** [reply]
oh, I see. I really do like ssd cache - makes my synology near silent in daily operations. I'll keep that in mind though when storage space becomes a constraint and a heavier priority than head-seeking noises.

**Comment by lastangel32** [comment]
Hi, but work on Apollolake cpu with 7.2.1 or letter?? Or only from 7.0.1 to 7.2?

I read this  “works in DSM 7.0.1 to 7.2 on models with one of following [CPU architectures]…, Apollolake…”

**Reply by 007revad** [reply]
Actually, I don't think it works at all for Apollolake. Let me check.

**Reply by 007revad** [reply]
DSM for models with an Apollolake CPU is missing 26 deduplication related files.

I've updated the readme.

**Comment by lastangel32** [comment]
> DSM for models with an Apollolake CPU is missing 26 deduplication related files.
> 
> I've updated the readme.

Perfect 

anyway, thanks for the work you do

**Comment by rbosch1887** [comment]
Working: ~no~  **YES**   (need to run "`syno_enable_dedupe.sh --hdd`" and "`synology_hdd_db -n`") as discussed below
NAS model: DS423+
DSM version: DSM 7.2.1-69057 Update 5
RAM: 18GB
SSD brand and model:  Samsung SSD 980 1TB as Volume 2 JBOD  with your Synology_M2_volume script
HDD: 4x HDD Seagate Ironwolf 4TB as Volume 1  SHR

grep -c 'SYNO.SDS.StorageUtils.supportBtrfsDedupe&&e.dedup_info.show_config_btn' /var/packages/StorageManager/target/ui/storage_panel.js
1
grep -c 'SYNO.SDS.StorageUtils.supportBtrfsDedupe,' /var/packages/StorageManager/target/ui/storage_panel.js
1
synogetkeyvalue /etc.defaults/synoinfo.conf support_btrfs_dedupe
yes
synogetkeyvalue /etc/synoinfo.conf support_btrfs_dedupe
yes
synogetkeyvalue /etc.defaults/synoinfo.conf support_tiny_btrfs_dedupe
no
synogetkeyvalue /etc/synoinfo.conf support_tiny_btrfs_dedupe
no

**Reply by 007revad** [reply]
Have you run https://github.com/007revad/Synology_HDD_db to add your drives to DSM's supported drive database? Run it with the -n option to prevent DSM updating the supported drive database.

Have you rebooted since running syno_enable_dedupe.sh ?

You could try running the script with the --hdd option
```
syno_enable_dedupe.sh --hdd
```

**Reply by rbosch1887** [reply]
yes,  but without the -n option. some months ago as i run the m2 volume script. an the scripts is scheduled at every shutdown.
but  will try running with the -n

yes rebooted twice after hour (waited the 20 minutes as stated somewhere up)

will try to run it with the --hdd option. 

will let you know

**Reply by rbosch1887** [reply]
> Have you run https://github.com/007revad/Synology_HDD_db to add your drives to DSM's supported drive database? Run it with the -n option to prevent DSM updating the supported drive database.
> 
> Have you rebooted since running syno_enable_dedupe.sh ?
> 
> You could try running the script with the --hdd option
> 
> ```
> syno_enable_dedupe.sh --hdd
> ```

it worked! right after reboot there is the deduplication menu item.
Thank you!

**Comment by StusBrainDump** [comment]
Hi, I see this is listed as working on 720+ 420+ 220+ but there is no mention of 920+ ? Anyone say it works?

**Reply by 007revad** [reply]
I'm not sure how I neglected to include the DS920+.

The script works on any model with a Geminilake CPU. I've added DS920+ to the "working list" now.

**Comment by lastangel32** [comment]
HI

Working: yes (no Automatic?)
NAS model: DS923+
DSM version: DSM 7.2.2-72806
SSD brand and model: Active on HDD driver (no ssd)

start manual deduplication fork fine but when enable "Automatic Decuplication" but it doesn't do anything (enable schedule data for all time of day)

![image](https://github.com/user-attachments/assets/0c268942-c2ae-4f96-80e2-d64e945ce32d)

**Reply by t0mm1e** [reply]
How do you know it isn't working?

**Reply by lastangel32** [reply]
Hi,
the script work perfect, thanks
eh...godd question...how i can find last deduplication state run?

i had tried 

~$ sudo btrfs version
btrfs-progs v4.0

~$ sudo btrfs dedupe status /volume3
btrfs: unknown token 'dedupe'

**Reply by t0mm1e** [reply]
Check ` /var/log/synobtrfsdedupe.log` I suppose it seems to contain reports from dedupe activity, including how much was cloned.

You can also check how much data is 'exclusive', not cloned that is:

```
root@HugeStorage:/volume1/Storage/Downloads# btrfs filesystem du --summarize 2024
     Total   Exclusive  Set shared  Filename
 245.14GiB    22.11GiB   186.44GiB  2024
```
(too bad no Markdown support inside code blocks, wanted to make the btrfs command bold so it would stand out more)

That is only 22 GiB of real data. But... That doesn't seem right, that's way too much non-exclusive.
I don't have much time left to check that out, but it's kind of important. If something is amiss with my storage or backups, that would be a disaster if something would happen and the data isn't really there to be restored in case of emergency.

**Reply by lastangel32** [reply]
hi, one easy question.
if i need to will replace the DiskStation i need to Undeduplication data before change it? (ipotize a replacement)

**Reply by 007revad** [reply]
No. Just migrate the drives to the new Synology.

**Comment by francoism90** [comment]
Working: yes
NAS model: DS923+
DSM version: 7.2.2-72806
SSD brand and model: Samsung 960 EVO 250 GB (2x - RAID 1)

It seems to be working:

![image](https://github.com/user-attachments/assets/1cdd00b7-3354-41d9-bf01-3ee49055f417)

I use `tiny` and did run your hardware allowlist updater. :)
For some reason it didn't work before, maybe because I didn't had enough space, and I didn't fully understood the `-n` flag for the HDD enabler. My HDD setup is 3x8TB, and 1x16TB (RAID-0, it's a backup for a backup).

It's running since yesterday, I'll report back when it's finished.

Thanks for your tools and support!

**Reply by francoism90** [reply]
![image](https://github.com/user-attachments/assets/f7b77901-d693-4b16-bbce-3b738c4c3f2b)

More space. :)

**Comment by kranix0** [comment]
**Working: ~~no~~ yes**
NAS model: DS-920+
DSM version: 7.2.2-72806 Update 2
HDD brand and model: WDC WD140EFDZ-11A0VA0

The "Configure data deduplication" option is not available.

I had to run syno_hdd_db.sh to remove the drive errors. No errors since then. I have set Task Scheduler to run syno_enable_dedupe.sh and syno_hdd_db.sh on shutdown.

**Reply by kranix0** [reply]
I spoke too soon. I realised that I had to run the script with `--hdd`, then it started working (after a reboot). I'm running the dedup analyser now.

**Reply by francoism90** [reply]
@kranix0 How much memory do you have? See the `--tiny` flag if you need it.

**Reply by kranix0** [reply]
Thanks for your reply. I've got 20GB RAM total, with a 2GB VM running as well.

The Deduplication Analyser is still at 0.0% after nearly an hour. I'll leave it running overnight and see what happens. If it hasn't made good progress, I'll try again with `--tiny` as you suggest.

**Reply by francoism90** [reply]
@kranix0 Ah, you don't seem to need the `--tiny` flag. I needed this before upgrading the memory. :)

It can take a lot of time, it took 2/3 days for me in total, if I can remember correctly.
Just to make sure, you have installed and Usage detail analysis enabled?

**Reply by kranix0** [reply]
Yes, it's taking a while: it's currently at 33%. Perhaps not surprising since my volume is pretty full: 1.9TB free out of 12.7TB total (85%).

I've got usage detail analysis enabled. I suppose I just have to be patient.

**Reply by 007revad** [reply]
By my calculations it has another 26 hours to go.

**Reply by kranix0** [reply]
I forgot to mention that the Analyser finished just fine. I haven't yet decided to proceed. I figured I'll set up volume encryption and maybe defragmentation first. I understand that my existing snapshots will become mucked up and new snapshots will be OK. Otherwise, tell me if I'm making a big mistake 🤣

**Reply by 007revad** [reply]
Defragging can actually increase the space used.

I would probably temporarily disable snapshots and delete all the existing snapshots. Then after deduping re-enable snapshots. But make sure you have an up to date backup first.

**Comment by Rhubarb1** [comment]
Please comment if this worked for you, or didn't work for you, with the following information:

    Working: yes  
    NAS model: xpenology 3622xs+ with 32GB ECC RAM, Intel Xeon E3-1245v6, and /volume1 of 55.8 TB (with 6 x Seagate Exos X16 16TB HDD in SHR-2)
    DSM version: DSM 7.2.1-69057 Update 6 
    SSD brand and model:  no ssd
Resource monitor indicated 3~4% cpu initially,subsequently reduced to 0.5% after a couple of hours.
Waiting to find out what actual deduplication is achieved.

**Comment by Rhubarb1** [comment]
The volume is more than 22TB.

**Reply by 007revad** [reply]
That will take days. Probably 2 more days to go.

**Comment by laurulbalaurul** [comment]
Working: yes
NAS model: DS723+
DSM version: DSM 7.2.2-72806 Update 3
SSD brand and model: WD Red SN700 500GB (2x500GB SHR RAID)

**Comment by Rhubarb1** [comment]
Working: Yes  (Was working initially under DSM7.2.1u4, until forced to do a reinstall some time later). Then wasn't working until tried again today.  
    NAS model: xPenology DS3622xs+ with Intel i7-14700K
    DSM version: DSM7.2.1u7
    SSD brand and model: No SSD's, 
           2xSHR2 volumes each 83.8TB and comprising 8xSeagate Exos X16 16TB 
Comment: When I ran the script:  "syno_enable_dedupe.sh --hdd", I was prompted to refresh the screen (and possibly reboot).  Storage Manager then showed the option to dedupe each volume (/volume1 and /volume2) with option of automatic or Manual.  
I selected Automatic and Resource Manager shows that deduplication is happening now:-)

**Comment by hawkeye73** [comment]
Working: Yes
NAS model: DS1821+
DSM Version: 7.3.1-86003 Update 1
SSD Brand and model: 2x 1 TB Seagate Firecuda 510
HDD Brand and model: 8x 16 TB Toshiba MG08ACA16TE

**Comment by Rhubarb1** [comment]
Latest update re my earlier posts of 30 January, 2025 and April 7.  Both systems are now configured as follows:
1. Intel Xeon e3-1245v6, 32GB ECC mem, 8x16TB HDDs in single SHR2 83.8TB volume @ 57% capacity (48.5TB); 35.3TB free. (My Test/Backup box; secondary Plex server, RR loader v25.9.7 / Synology DSM 7.3.1-86003). Total data stored in Shared folders = 87.3TB. Dedupe savings: 38.8TB
2. Intel i7-14700k, 64GB ECC mem, 12x16TB HDDs in single SHR2 139.6TB volume @ 51% capacity (72.4TB); 67.2TB free. (Primary NAS system, Plex server & off-site backup server, RR v25.8.1 / DSM 7.3-81180). Total data stored in shared folders 90.2TB.  Dedupe savings: 18TB

**Comment by babujnik** [comment]
Working: No
NAS model: DS725+
DSM version: DSM 7.3.2-86009
SSD brand and model:  2x 512G GoodRam SSD
HDD Brand and model: 2x 4TB WD Red

script shows:

`NAS has 4 GB of memory.

Non-Synology drive support already enabled.

Tiny Btrfs Data Deduplication already enabled.

Dedupe config menu for HDDs and 2.5" SSDs is enabled.

Finished
`

Yet no dedup option visible

**Comment by 007revad** [comment]
Hi @babujnik 
It was working on my DS925+ with DSM 7.3 and 7.3.1 but it's not working now with 7.3.2. I'll have a look at what's changed.

**Reply by OdinochkaAO** [reply]
Oh, and I was thinking where did I screw up, thanks for the info! Been trying to find out whats the issue for 2 days. Doesn't work either. There is a menu item for dedup but it doesn't actually work. Looks like it doesn't see the volume data. Shows saved space = 0 if triggered manually.  Here are the parameters and logs if necessary:
```
root@syno-925:~# grep -c 'SYNO.SDS.StorageUtils.supportBtrfsDedupe&&e.dedup_info.show_config_btn' /var/packages/StorageManager/target/ui/storage_panel.js
0
root@syno-925:~# grep -c 'SYNO.SDS.StorageUtils.supportBtrfsDedupe,' /var/packages/StorageManager/target/ui/storage_panel.js
1
root@syno-925:~# synogetkeyvalue /etc.defaults/synoinfo.conf support_btrfs_dedupe
yes
root@syno-925:~# synogetkeyvalue /etc/synoinfo.conf support_btrfs_dedupe
yes
root@syno-925:~# synogetkeyvalue /etc.defaults/synoinfo.conf support_tiny_btrfs_dedupe
no
root@syno-925:~# synogetkeyvalue /etc/synoinfo.conf support_tiny_btrfs_dedupe
no
```

Log:
```
2026-01-31T14:44:21+03:00 syno-925 synobtrfsdedupe[2522]: ----- Init new data dedupe workflow: /volume1 -----
2026-01-31T14:44:22+03:00 syno-925 synobtrfsdedupe[2522]: Total data scanned         : 1.16     MB
2026-01-31T14:44:22+03:00 syno-925 synobtrfsdedupe[2522]: Hole                       : 0.00     B
2026-01-31T14:44:22+03:00 syno-925 synobtrfsdedupe[2522]: Small extent               : 172.00   KB
2026-01-31T14:44:22+03:00 syno-925 synobtrfsdedupe[2522]: Cloned extent              : 0.00     B
2026-01-31T14:44:22+03:00 syno-925 synobtrfsdedupe[2522]: Rewrote extent             : 0.00     B
2026-01-31T14:44:22+03:00 syno-925 synobtrfsdedupe[2522]: Not found error count      : 0
2026-01-31T14:44:22+03:00 syno-925 synobtrfsdedupe[2522]: Dedupe :
2026-01-31T14:44:22+03:00 syno-925 synobtrfsdedupe[2522]: Extent-same requested      : 0.00     B (0)
2026-01-31T14:44:22+03:00 syno-925 synobtrfsdedupe[2522]: Success                    : 0.00     B (0)
2026-01-31T14:44:22+03:00 syno-925 synobtrfsdedupe[2522]: Zero                       : 0.00     B (0)
2026-01-31T14:44:22+03:00 syno-925 synobtrfsdedupe[2522]: Failed by diff data        : 0.00     B (0)
2026-01-31T14:44:22+03:00 syno-925 synobtrfsdedupe[2522]: Failed by toxic            : 0.00     B (0)
2026-01-31T14:44:22+03:00 syno-925 synobtrfsdedupe[2522]: Failed by src not fount    : 0.00     B (0)
2026-01-31T14:44:22+03:00 syno-925 synobtrfsdedupe[2522]: Failed by dst not fount    : 0.00     B (0)
2026-01-31T14:44:22+03:00 syno-925 synobtrfsdedupe[2522]: Failed by other error      : 0.00     B (0)
2026-01-31T14:44:22+03:00 syno-925 synobtrfsdedupe[2522]: Too small                  : 0.00     B (0)
2026-01-31T14:44:22+03:00 syno-925 synobtrfsdedupe[2522]: Success ranges :
2026-01-31T14:44:22+03:00 syno-925 synobtrfsdedupe[2522]: 128K - 256K                : 0.00     B
2026-01-31T14:44:22+03:00 syno-925 synobtrfsdedupe[2522]: 256K - 512K                : 0.00     B
2026-01-31T14:44:22+03:00 syno-925 synobtrfsdedupe[2522]: 512K - 1M                  : 0.00     B
2026-01-31T14:44:22+03:00 syno-925 synobtrfsdedupe[2522]: 1M   - 4M                  : 0.00     B
2026-01-31T14:44:22+03:00 syno-925 synobtrfsdedupe[2522]: 4M   - 16M                 : 0.00     B
2026-01-31T14:44:22+03:00 syno-925 synobtrfsdedupe[2522]: 16M  - 64M                 : 0.00     B
2026-01-31T14:44:22+03:00 syno-925 synobtrfsdedupe[2522]: 64M  -                     : 0.00     B
2026-01-31T14:44:22+03:00 syno-925 synobtrfsdedupe[2522]: Release by extent-same     : 0.00     B
2026-01-31T14:44:22+03:00 syno-925 synobtrfsdedupe[2522]: Release by reclaim         : 0.00     B
2026-01-31T14:44:22+03:00 syno-925 synobtrfsdedupe[2522]: Total release              : 0.00     B
2026-01-31T14:44:22+03:00 syno-925 synobtrfsdedupe[2522]: Sample hit                 : 0
2026-01-31T14:44:22+03:00 syno-925 synobtrfsdedupe[2522]: Load bucket                : 0
2026-01-31T14:44:22+03:00 syno-925 synobtrfsdedupe[2522]: Cache hit                  : 0
2026-01-31T14:44:22+03:00 syno-925 synobtrfsdedupe[2522]: Record rotated             : 0
2026-01-31T14:44:22+03:00 syno-925 synobtrfsdedupe[2522]: Rotated fp size            : 0.00     B
2026-01-31T14:44:22+03:00 syno-925 synobtrfsdedupe[2522]: ----- End  new data dedupe workflow: /volume1 -----
2026-01-31T14:52:03+03:00 syno-925 synobtrfsdedupe[2522]: Load sample index successfully
2026-01-31T14:52:03+03:00 syno-925 synobtrfsdedupe[2522]: ----- Init light dedupe workflow: /volume1 -----
2026-01-31T14:52:03+03:00 syno-925 synobtrfsdedupe[2522]: No data scanned
2026-01-31T14:52:03+03:00 syno-925 synobtrfsdedupe[2522]: ----- End  light dedupe workflow: /volume1 -----

```

Device: DS 925+
RAM: Samsung ECC 32Gb (2x16)
HDD: WD DC560 20Tb x4
SSD Cache: Yes, Samsung 990 Pro
DSM: 7.3.2 U1

**Comment by ghost** [comment]


**Reply by mcfaul** [reply]
Exactly the same here on my DS923+ (also DSM 7.3.2-86009, but I am on update 3), everything seems to work and the script tells me its all enabled, but I don't see the dedupe option in the menu?

Non-Synology drive support already enabled.

Tiny Btrfs Data Deduplication already enabled.

Dedupe config menu for HDDs and 2.5" SSDs already enabled.

**Comment by Rhubarb1** [comment]
Since my last reply on 19 November, 2025, there have been a couple of hiccups where my disks were marked as unverified. As a result I was forced to run the scripts again on my two systems.  Both systems are running DSM 7.3.2-86009 Update3 with deduplication working correctly. Main box: Intel i7-14700K, Asus Pro WS W680-ACE with 64GB ECC RAM, in a single SHR2 /volume1 (12x 16TB (+1 hot spare) Seagate Exos X16, now 139.6TB experienced more troubles than my second box: Intel Xeon E3-1245v6 in Supermicro m/brd X11SSL-cf with 32GB ECC RAM 8x16TB Exos X16 
<img width="1491" height="965" alt="Dumbo Dedup 12x16TBSeagateEXOSx16-139 6TB volume1 22 04 2026" src="https://github.com/user-attachments/assets/e945d465-2b0e-49cb-8902-be2237a409c6" />
<img width="1567" height="844" alt="Dupey Dedup 8x16TBSeagateEXOSx16-87 3TB volume1 22 04 2026" src="https://github.com/user-attachments/assets/91d066c1-269f-41c5-881f-3e94cb65c759" />

**Comment by Glennada** [comment]
Not working as yet..

Model: DS1520+
CPU: Intel Celeron J4125 (Geminilake)
DSM: 7.3.2-86009
RAM: 8GB
Flag confirmed set: support_tiny_btrfs_dedupe="yes" in both /etc/synoinfo.conf and /etc.defaults/synoinfo.conf
UI still not showing dedup tab in Storage Manager

**Comment by hawkeye73** [comment]
Working: Yes
NAS model: DS1821+
DSM Version: 7.4-90075
SSD Brand and model: 2x 1 TB Seagate Firecuda 510
HDD Brand and model: 8x 16 TB Toshiba MG08ACA16TE

<img width="293" height="236" alt="image" src="https://github.com/user-attachments/assets/b0106ecd-5367-4477-8f89-56c25d64c97f" />

After waiting some minutes I can choose the first one which enable which shared folders will use dedup and/or compression. The second one enables/disable the actual deduping

**Comment by cb12tre** [comment]
Working: partially
NAS model: DS920+
DSM version: 7.4-90075
HDD: WDC WD60EFRX
SSD brand and model: SAMSUNG MZALQ512HALU-000L1 and SOLIDIGM SSDPFKNU010TZH

```
sudo -s ./syno_enable_dedupe.sh -c
Password:
Synology_enable_Deduplication v1.4.33
DS920+ DSM 7.4-90075

StorageManager 1.1.0-2018

Using options: -c
Running from: /volume1/scripts/Synology_enable_Deduplication/syno_enable_dedupe.sh

NAS has 8 GB of memory.

Btrfs Data Deduplication is not enabled.

Tiny Btrfs Data Deduplication is enabled.

Dedupe config menu for HDDs and 2.5" SSDs is enabled.

Checking non-Synology drive supported.
File is already edited.

Checking value in backup file.
Backup file is okay.
```

I can't see "Configure Storage Efficiency:

<img width="825" height="273" alt="image" src="https://github.com/user-attachments/assets/004168ae-c3fe-402e-8ece-7c09608bc8d0" />


Deduplication appears to be enabled. I enabled it about a year ago on an older version. As you can see, it doesn't allow you to disable "Usage detail analysis"

<img width="521" height="358" alt="image" src="https://github.com/user-attachments/assets/57ecf2bd-da8e-43aa-b485-1d6957d3d550" />

**Comment by hawkeye73** [comment]
Working: yes (only duplicate menu)
NAS model: DS920+ (20 GB RAM)
DSM version: 7.4-90075
HDD brand and model:
ST8000VN004
ST4000DM000
ST8000VN0022
HUH728080ALE601

**Comment by jivemaster2005** [comment]
Working: yes
NAS model: DS1621+ (32GB RAM)
DSM version: 7.4-90075
HDD: WDC WD120EMFZ, WD120EMAZ. Seagate ST14000NE0008 (in a DX517)
SSD brand and model: SAMSUNG SSD 990 PRO 2TB

**Comment by Startric** [comment]
**Working:** Yes

**NAS model:** Synology DS423+

**DSM version:** 7.4.1-90080 (last)

**HDD (Volume 1):** 4x WD Red Plus — 2x WD40EFRX-68N32N0 + 2x WD40EFPX-68C6CN0 (4TB each, status Healthy)

**SSD/NVMe (Volume 2):** 2x WD Red SN700 1TB (M.2 NVMe, status Healthy)

Thanks !
